controllers/drplacementcontrol.go

```diff
@@ -367,6 +382,32 @@ func (d *DRPCInstance) RunFailover() (bool, error) {
 	return d.switchToFailoverCluster()
 }
 
+// isValidFailoverTarget determines if the passed in cluster is a valid target to failover to. A valid failover target
+// may already be Primary, if it is Secondary then it has to be protecting PVCs with VolSync.
+// NOTE: Currently there is a gap where, right after DR protection when a Secondary VRG is not yet created for VolSync
+// workloads, a failover if initiated will pass these checks. When we fix to retain VRG for VR as well, a more
+// deterministic check for VRG as Secondary can be performed.
+func (d *DRPCInstance) isValidFailoverTarget(cluster string) bool {
+	vrg, ok := d.vrgs[cluster]
```

Comment:
This function may incorrectly declare the cluster as healthy due to failures in retrieving the VRG from it.

Reply:
Good catch, only way out I see is to call `GetVRGFromManagedCluster` and ignore NotFound error and ensure we found a VRG or not. Stashed VRGs in DRPCInstance will not help with the error. Any other thoughts?

Reply:
Maybe go the long way. Change the d.vrgs to be a
```
map[string]*VRGQuery
```
where the VRGQuery is type of
```
type VRGQuery struct {
  vrg *rmn.VolumeReplicationGroup
  requestResult bool
}
```
Where `requestResult` is either true or false. When true, it means successfully queried the cluster, the vrg pointer determines whether the VRG was found or not.  If false, it means the request failed and the `vrg` field is **guaranteed** to be `nil`.

I am not sure if you want to do that. However, in the long term, we should actually refrain from calling `GetVRGFromManagedCluster` from all over the place (unfortunately we do today).  Each reconciliation we populate `d.args` and we should use that array throughout.

Reply:
Agreed, for now I am going to use the shortcut (the other change is quite intrusive) and follow up with a broader change to leverage d.vrgs with the required information.

Reply:
Issue to track using cached VRGs opened here: https://github.com/RamenDR/ramen/issues/1384